### DISCUSSION TITLE
Userscript for highlighting text from title

### DISCUSSION THREAD

**Discussion by Heholord** [sebastian-hofstaetter/teaching#55] [Show and tell]
Dear collegues, 

I wrote a script that highlights the words from the title within the body text. As most of you have probably found out, often the interesting parts will appear close to these highlights.
The script is not perfect but it helps. If you come up with some improvements, or you can get rid of the forceful page reloads don't hesitate to create a pull-request. 
The link to the script is: https://github.com/Heholord/FIRA_Userscript_Highligter

Here is an example of how it will look like: 
![screenshot-2022-04-03_19-47-39](https://user-images.githubusercontent.com/2659160/161446022-06063ac1-6fd2-4dca-b1f8-cc9657cda319.png)

Hopefully you will see this screen soon:
![screenshot-2022-04-03_21-47-18](https://user-images.githubusercontent.com/2659160/161446026-a4d2618c-51c6-44a6-b793-de9a5b908e87.png)

Best

**Comment by Heholord** [comment]
Thanks to Bernhard (https://github.com/bernikr) we now do not reload the page when the next article is loaded (or the tagging-buttons have been pressed).

**Comment by sebastian-hofstaetter** [comment]
Hi,
Pretty cool that this works. **However, I have to please ask everyone to not actively use this tool.** The reason for that is by using this tool you are changing the setup of the annotation task, which will make the data more noisy, and we might take the wrong conclusions from it, when we assume everyone annotated the same screen (without highlighting). 

It is pretty interesting to think about why highlighting the key terms makes the task easier - both for humans as well as the neural models (given how that basically boils down to term-matching). For exercise 2 and beyond we are interested in evaluating neural models, that can cross this boundary of pure exact term-matching, therefore it is vital to have unbiased data. Using the highlighting very likely biases annotators towards higher labels when they see more term matches, which reduces the need for semantic similarities beyond them (which is a completely valid outcome and interesting study if we could control for it = we know which samples have been highlighted and which haven't, but we don't know in this case). We'll definitely take this as an idea to integrate into fira in the future, but for now, please don't use it. Thank you!

Best,
Sebastian

**Reply by Heholord** [reply]
I will make this project private to support what has been stated. I also thought about how the outcome might become biased when I wrote this script, so that's that. 
I will make it public again after the assignment. PM me if you feel the need to discuss this decision.

**Reply by Heholord** [reply]
It is now public again
